UpstreamTagger/UpstreamTaggerHit.cxx

```diff
@@ -30,21 +30,22 @@ UpstreamTaggerHit::UpstreamTaggerHit()
 
 
 // -----   constructor from TimeDetPoint from TimeDetHit-------------------------------
-UpstreamTaggerHit::UpstreamTaggerHit(UpstreamTaggerPoint* p, Double_t t0)
+UpstreamTaggerHit::UpstreamTaggerHit(UpstreamTaggerPoint* p, UpstreamTagger* c, Double_t t0)
```

Comment:
This should probably be a const reference.

Reply:
It works fine as a pointer.

Reply:
It may work, but this way it allows (accidental) modification of the UpstreamTagger by the hit. As there is no good reason I can think of for this to happen, it's useful to have the compiler enforce this by making it const.

Whether it's a const reference or const pointer, I frankly don't care, but conventionally const reference are usually used for this purpose.